crates/polars-time/src/chunkedarray/rolling_window/rolling_kernels/no_nulls.rs

```diff
@@ -50,6 +53,75 @@ where
     Ok(Box::new(out))
 }
 
+// Use an aggregation window that maintains the state
+pub(crate) fn rolling_apply_agg_window<'a, Agg, T, O>(
+    values: &'a [T],
+    offsets: O,
+    min_periods: usize,
+    params: DynArgs,
+    sorting_indices: Option<&[IdxSize]>,
+) -> PolarsResult<ArrayRef>
+where
+    // items (offset, len) -> so offsets are offset, offset + len
+    Agg: RollingAggWindowNoNulls<'a, T>,
+    O: Iterator<Item = PolarsResult<(IdxSize, IdxSize)>> + TrustedLen,
+    T: Debug + IsFloat + NativeType,
+{
+    if values.is_empty() {
+        let out: Vec<T> = vec![];
+        return Ok(Box::new(PrimitiveArray::new(
+            T::PRIMITIVE.into(),
+            out.into(),
+            None,
+        )));
+    }
+    let sorting_indices = sorting_indices.expect("`sorting_indices` should have been set");
+    // start with a dummy index, will be overwritten on first iteration.
+    let mut agg_window = Agg::new(values, 0, 0, params);
+
+    let mut out = zeroed_vec(values.len());
+    let mut null_positions = Vec::with_capacity(values.len());
```

Comment:
Should this allocate this much?

Reply:
nope, just pushing an update

Reply:
Just chatted with Orson about this

The difficulty is that it's not known in advance (before starting the loop) whether there will be nulls
He's suggested making an `Option<MutableBitmap>`, which gets set on the first null

There is  an extra `if validity.is_none()` check, but nulls produced by the rolling operation should be fairly rare anyway

I've pushed an update - thanks @orlp, much appreciated 🙏